synapse/app/homeserver.py

```diff
@@ -655,12 +655,14 @@ def start_generate_monthly_active_users():
             "generate_monthly_active_users", generate_monthly_active_users
         )
 
-    start_generate_monthly_active_users()
-    if hs.config.limit_usage_by_mau or hs.config.mau_stats_only:
+    if hs.config.run_background_tasks and (
+        hs.config.limit_usage_by_mau or hs.config.mau_stats_only
+    ):
+        start_generate_monthly_active_users()
         clock.looping_call(start_generate_monthly_active_users, 5 * 60 * 1000)
     # End of monthly active user settings
 
-    if hs.config.report_stats:
+    if hs.config.run_background_tasks and hs.config.report_stats:
```

Comment:
Do we not need to add these to the `generic_worker` so that they can run on workers?

Reply:
Good call! I moved the phone home stats code out to a separate module and now instantiate it for both this or a generic worker. The code was pretty much copied and pasted in b226c49dc5e6722724e85438fe6a07e12ca28d41.